shims/direnv_use_asdf

```diff
@@ -102,58 +102,65 @@ _all_plugins_list() {
 }
 
 _except_local_plugins_list() {
-  local tool_versions=$1
-  local tmp_local_plugin_names
+  tool_versions=$1
   tmp_local_plugin_names="$(mktemp)"
   _plugins_in_file "$tool_versions" >"$tmp_local_plugin_names"
   _all_plugins_list | grep -v -F -f "$tmp_local_plugin_names"
-  rm "$tmp_local_plugin_names"
+  /bin/rm "$tmp_local_plugin_names"
 }
 
 _load_global_plugins_env() {
-  local tool_versions=$1
+  tool_versions=$1
   _except_local_plugins_list "$tool_versions" | _tail_r | _pardo _load_plugin_version_and_file
 }
 
 _load_local_plugins_env() {
-  local tool_versions=$1
+  tool_versions=$1
   _plugins_in_file "$tool_versions" | _tail_r | _pardo _load_plugin_version_and_file
 }
 
 _plugin_env() {
-  asdf exec direnv dotenv bash <(env)
+  used_shell=$(echo "$SHELL" | rev | cut -d '/' -f1 | rev)
+  temp_env_file=$(mktemp)
+  env >"$temp_env_file"
+  asdf exec direnv dotenv "$used_shell" "$temp_env_file"
+  /bin/rm "$temp_env_file"
 }
 
 # from asdf plugin_current_command
 _load_plugin_version_and_file() {
-  local plugin_name=$1
-  local versions_and_path
+  plugin_name=$1
   versions_and_path="$(find_versions "$plugin_name" "$(pwd)")"
   if test -z "$versions_and_path"; then
     return 0
   fi
 
-  local path
-  path=$(cut -d '|' -f 2 <<<"$versions_and_path")
-  IFS=$'\n' read -r -a versions <<<"$(cut -d '|' -f 1 <<<"$versions_and_path" | uniq)"
-  for version in "${versions[@]}"; do
+  # path=$(cut -d '|' -f 2 <<<"$versions_and_path")
+  path=$(printf '%s\n' "$versions_and_path" | cut -d '|' -f 2)
+  unique_versions=$(printf '%s\n' "$versions_and_path" | cut -d '|' -f 1 | uniq)
+  # IFS=$'\n' read -r -a versions <<<"$unique_versions"
+  for version in ${unique_versions}; do
     _plugin_env_bash "$plugin_name" "$version"
     log_status "using asdf ${plugin_name} ${version}"
   done
   if [ -f "$path" ]; then
-    asdf exec direnv watch bash "$path"
+    used_shell=$(echo "$SHELL" | rev | cut -d '/' -f1 | rev)
+    asdf exec direnv watch "$used_shell" "$path"
   fi
 }
 
 _plugin_env_bash() {
-  local plugin="${1}"
-  local version="${2}"
-  local old_path="$PATH"
+  plugin="${1}"
+  version="${2}"
+  old_path="$PATH"
   if [ -n "$version" ]; then
     with_plugin_env "$plugin" "$version" _plugin_env | sed "s#:$old_path#:'\"\$PATH\"'#"
   fi
 }
 
-if [ "do" == "$1" ]; then
-  eval "${@:2}"
+if [ "do" = "$1" ]; then
+  shift 1
+  eval "${@}"
 fi
+
+set +e
```

Comment:
Let's remove this `set +e`. We should avoid changing those flags unless explicitly needed and even then, restore to their original value.

Reply:
@vic We need to keep this flag, because either the `. .../utils.sh` or the `eval .... stdlib` causes the script to propagate `set -e`
At least that was the case in my terminal.

Reply:
Looks like `set -e` is actually set by direnv itself, see first lines of `direnv stlib`.

You might want to add `set +e` in your local `.envrc` manually or in `~/.config/direnv/direnvrc` to set it globally.